src/Conda.jl

```diff
@@ -35,42 +35,110 @@ using Compat
+    if !isdir(path)
+        throw(ArgumentError("Path to conda environment is not valid."))
+    end
+    return path
+end
+
+if !isdir(ROOTENV)
+    # Ensure ROOTENV exists, otherwise defining constants below will throw errors
+    mkpath(ROOTENV)
+end
+
+const PREFIX = prefix(ROOTENV)
 
 "Prefix for the executable files installed with the packages"
-const BINDIR = is_windows() ? joinpath(PREFIX, "Library", "bin") : joinpath(PREFIX, "bin")
+function bin_dir(env::Environment)
+    return is_windows() ? joinpath(prefix(env), "Library", "bin") : joinpath(prefix(env), "bin")
+end
+const BINDIR = bin_dir(ROOTENV)
+
+"Prefix for the shared libraries installed with the packages"
+function lib_dir(env::Environment)
+    return is_windows() ? joinpath(prefix(env), "Library", "bin") : joinpath(prefix(env), "lib")
+end
+const LIBDIR = lib_dir(ROOTENV)
 
 "Prefix for the python scripts. On UNIX, this is the same than Conda.BINDIR"
-const SCRIPTDIR = is_windows() ? joinpath(PREFIX, "Scripts") : BINDIR
+function script_dir(env::Environment)
+    return is_windows() ? joinpath(prefix(env), "Scripts") : bin_dir(env)
+end
+const SCRIPTDIR = script_dir(ROOTENV)
 
 "Prefix where the `python` command lives"
-const PYTHONDIR = is_windows() ? PREFIX : BINDIR
+function python_dir(env::Environment)
+    return is_windows() ? prefix(env) : bin_dir(env)
+end
+const PYTHONDIR = python_dir(ROOTENV)
 
-"Path to the `conda` binary"
-const conda = joinpath(SCRIPTDIR, "conda")
+function conda_bin(env::Environment)
+    if is_windows()
+        p = script_dir(env)
+        conda_bat = joinpath(p, "conda.bat")
+        return isfile(conda_bat) ? conda_bat : joinpath(p, "conda.exe")
+    else
+        return joinpath(script_dir(env), "conda")
+    end
+end
+const conda = conda_bin(ROOTENV)
 
 "Path to the condarc file"
 const CONDARC = joinpath(PREFIX, "condarc-julia")
 
+
 """
 Use a cleaned up environment for the command `cmd`.
 
 Any environment variable starting by CONDA will interact with the run.
 """
-function _set_conda_env(cmd)
-    env = copy(ENV)
+function _set_conda_env(cmd, env::Environment=ROOTENV)
+    env_var = copy(ENV)
     to_remove = AbstractString[]
-    for var in keys(env)
+    for var in keys(env_var)
         if startswith(var, "CONDA")
             push!(to_remove, var)
         end
     end
     for var in to_remove
-        pop!(env, var)
+        pop!(env_var, var)
     end
-    env["CONDARC"] = CONDARC
-    setenv(cmd, env)
+    env_var["CONDARC"] = joinpath(prefix(env), "condarc-julia")
+    env_var["CONDA_PREFIX"] = env_var["CONDA_DEFAULT_ENV"] = prefix(env)
+    setenv(cmd, env_var)
+end
+
+"Run conda command with environment variables set."
+function runconda(args::Cmd, env::Environment=ROOTENV)
+    _install_conda(env)
+    run(_set_conda_env(`$(conda_bin(env)) $args`, env))
+end
+
+"Run conda command with environment variables set and return the output as a string"
+function readconda(args::Cmd, env::Environment=ROOTENV)
```

Comment:
Since this is only used with `JSON.parse`, maybe it should be called `parseconda` and be defined as
```jl
JSON.parse(readstring(_set_conda_env(`$(conda_bin(env)) $args --json`, env)))
```